scripts/setup-db-folder.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/sh
+
+log_any () {
+	echo -e "[\e[$1m$2\e[0m] $3"
+}
+
+log_error   () { log_any "31" "-" "$1"; }
+log_ok      () { log_any "32" "+" "$1"; }
+log_warning () { log_any "33" "~" "$1"; }
+log_hint    () { log_any "34" "?" "$1"; }
+log_info    () { log_any "35" "i" "$1"; }
+
+check_pwd () {
+  root="$(dirname $(realpath $(dirname "$0")))"
+  [ "$root" != "$(pwd)" ] && {
+    log_error "the script should be run from the root of the repo"
+    log_error "expected '$root'"
+    log_error "found '$(pwd)'"
+    exit 1
+  }
+}
+
+setup_folder () {
+    local folder="$1"
+
+    mkdir -p $folder/dev
+    mkdir $folder/prod
+    chmod -R 774 $folder
+    log_info "sudo permission needed to chown newly created '$folder'"
+    sudo chown -R :root $folder
```

Comment:
I think I am missing something but why do we need to `chown` by root ?

Reply:
really good question!
> **Note**
> however we don't really `chown` it, we only change the group ownership tks to the `:`
> maybe I should have used `chgrp` instead of `chown`

This directory, `mongodb` and its subdirs `dev` and `prod` will be mounted in the containers so that the DB is persistent when the container is killed and restarted. You also need to know that when a file/dir is mounted in a container its UID/GID and permissions are kept.

Reading the bitnami image (the one we use) [doc](https://docs.bitnami.com/tutorials/bitnami-best-practices-hardening-containers/) (section `Use arbitrary UUIDs`) you can learn that they are **non-root images**
Meaning that the process inside is run as an unprivileged user *(the 1001 precisely, this obviously adds a good security layer and made me open #98)*, so he won't be able to read/write our mounted volume :confused: 
That's why it **belongs to the root group**, and why we need to change our mounted folder permissions: so that the mongdb process inside the container can write into this directory!

Reply:
Thanks a lot for your response ! Makes sens to me now. I resolve this thread